protocol/nats_jetstream/v2/receiver.go

```diff
@@ -117,14 +160,53 @@ func NewConsumerFromConn(conn *nats.Conn, stream, subject string, jsmOpts []nats
 	return c, nil
 }
 
+// NewConsumerFromConnV2 consumes from filterSubjects given in ConsumerConfig or OrderedConsumerConfig
+// embedded in ConsumerOption. See: WithConsumerConfig(...) and WithOrderedConsumerConfig(...)
+func NewConsumerFromConnV2(conn *nats.Conn, jsOpts []jetstream.JetStreamOpt, opts ...ConsumerOption) (*Consumer, error) {
+	js, err := jetstream.New(conn, jsOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Consumer{
+		Receiver:      *NewReceiver(),
+		Conn:          conn,
+		JetStream:     js,
+		internalClose: make(chan struct{}, 1),
+	}
+
+	err = c.applyOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
 func (c *Consumer) OpenInbound(ctx context.Context) error {
 	c.subMtx.Lock()
 	defer c.subMtx.Unlock()
 
-	// Subscribe
-	sub, err := c.Subscriber.Subscribe(c.Jsm, c.Subject, c.MsgHandler, c.SubOpt...)
-	if err != nil {
-		return err
+	var sub *nats.Subscription
+	var consumeContext jetstream.ConsumeContext
+	var err error
+	if c.Jsm != nil {
+		// Subscribe
+		sub, err = c.Subscriber.Subscribe(c.Jsm, c.Subject, c.MsgHandler, c.SubOpt...)
+		if err != nil {
+			return err
+		}
+	} else if c.JetStream != nil {
+		err = c.createConsumer(ctx)
+		if err != nil {
+			return err
+		}
+		consumeContext, err = c.JetstreamConsumer.Consume(c.MsgHandlerV2, c.PullConsumeOpt...)
+		if err != nil {
+			return err
+		}
+	} else {
+		return ErrNoJetstream
```

Comment:
can be simplified to:

- just `if c.JetStream != nil {...}`
- remove final `else` and return error directly

Reply:
(avoids else logic)